uni/lib/controller/parsers/parser_courses.dart

```diff
@@ -1,38 +1,80 @@
-import 'dart:collection';
-
 import 'package:html/parser.dart' show parse;
 import 'package:http/http.dart' as http;
+import 'package:uni/model/entities/course.dart';
 
-Map<String, String> parseMultipleCourses(List<http.Response> responses) {
-  final Map<String, String> coursesStates = HashMap();
+List<Course> parseMultipleCourses(List<http.Response> responses) {
+  final List<Course> courses = [];
   for (var response in responses) {
-    final map = parseCourses(response);
-    coursesStates.addAll(map);
+    final c = parseCourses(response);
+    courses.addAll(c);
   }
-  return coursesStates;
+  return courses;
 }
 
-/// Extracts a map containing information about the user's courses from an HTTP
-/// [response].
-///
-/// *Note:*
-/// * a key in this map is the name of a course
-/// * a value in this map is the state of the corresponding course
-Map<String, String> parseCourses(http.Response response) {
+List<Course> parseCourses(http.Response response) {
   final document = parse(response.body);
+  final List<Course> courses = [];
 
-  final Map<String, String> coursesStates = HashMap();
+  String? faculty = response.request?.url.toString();
+  faculty =
+      faculty?.substring(faculty.indexOf('up.pt/')).replaceFirst('up.pt/', '');
+  faculty = faculty?.substring(0, faculty.indexOf('/')).trim();
 
```

Comment:
Could be a faculty.split("/")[3] or faculty.split("/")[2] i don't know how it handles on '//'

Reply:
Assuming that we cannot be absolutely sure that the `http://` is present on the url, I think it is safer (although uglier) this way